sgkit/stats/conversion.py

```diff
@@ -0,0 +1,126 @@
+import dask.array as da
+import numpy as np
+from numba import guvectorize
+from xarray import Dataset
+
+from sgkit import variables
+from sgkit.typing import ArrayLike
+from sgkit.utils import conditional_merge_datasets
+
+
+@guvectorize(  # type: ignore
+    [
+        "void(float64[:], uint8[:], float64, int8[:])",
+        "void(float32[:], uint8[:], float64, int8[:])",
+    ],
+    "(p),(k),()->(k)",
+    nopython=True,
+    cache=True,
+)
+def _convert_probability_to_call(
+    gp: ArrayLike, _: ArrayLike, threshold: float, out: ArrayLike
+) -> None:  # pragma: no cover
+    """Generalized U-function for converting genotype probabilities to hard calls
+
+    Parameters
+    ----------
+    gp
+        Genotype probabilities of shape (genotypes,) containing unphased, biallelic
+        probabilities in the order homozygous reference, heterozygous, homozygous alternate.
+    _
+        Dummy variable of type `uint8` and shape (ploidy,) used to define
+        the ploidy of the resulting array
+    threshold
+        Probability threshold that must be met or exceeded by at least one genotype
+        probability in order for any calls to be made -- all values will be -1 (missing)
+        otherwise. Setting this value to less than 0 disables any effect it has.
+    out
+        Hard calls array of shape (ploidy,).
+    """
+    # Ignore singleton array inputs used for metadata inference by dask
+    if gp.shape[0] == 1 and out.shape[0] == 1:
+        return
+    if gp.shape[0] != 3 or out.shape[0] != 2:
+        raise NotImplementedError(
+            "Hard call conversion only supported for diploid, biallelic genotypes."
+        )
+    out[:] = -1  # (ploidy,)
+    # Return no call if any probability is absent
+    if np.any(np.isnan(gp)):
+        return
+    i = np.argmax(gp)
+    # Return no call if max probability does not exceed threshold
+    if threshold > 0 and gp[i] < threshold:
+        return
+    # Return no call if max probability is not unique
+    if (gp[i] == gp).sum() > 1:
+        return
+    # Homozygous reference
+    if i == 0:
+        out[:] = 0
+    # Heterozygous
+    elif i == 1:
+        out[0] = 1
+        out[1] = 0
+    # Homozygous alternate
+    else:
+        out[:] = 1
+
+
+def convert_probability_to_call(
```

Comment:
Do you want to add this to the public API docs?

Reply:
https://github.com/pystatgen/sgkit/pull/419/commits/61176be7f6d943ee12d28057086f73c13d3fdb5b#diff-0ee295c29def16f570f86378eff8ba37315e464d415e731d5d8ea2e67ce16f68R45

Is there any more to it?

Reply:
That should do it (you can check by generating the docs).